### ISSUE TITLE
[a11y]: On text spacing, text gets truncated

### ISSUE THREAD

**Issue by saiteja-nizam** [carbon-design-system/carbon#22125]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.102.0

### Automated testing tool and ruleset

Text spacing editor

### Description

on text spacing, In Dropdown or ComboBox components list items text gets truncated if its length is long.

<img width="1728" height="831" alt="Image" src="https://github.com/user-attachments/assets/dbdf397b-1491-4a1a-8152-d862a47dd44e" />
<img width="1724" height="954" alt="Image" src="https://github.com/user-attachments/assets/8ddc5831-97a6-4298-8c25-65285a933489" />

### WCAG 2.1 Violation

1.4.12 Text Spacing

### Reproduction/example

https://stackblitz.com/run?file=src%2FApp.jsx

### Steps to reproduce

Open stackblitz template for the Dropdown or ComboBox and give option text lengthier one, the text gets truncated.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@saiteja-nizam That stackblitz link doesn't work for me, but per the [design spec](https://carbondesignsystem.com/components/dropdown/usage/#overflow-content) this is working as expected
